main.tf

```diff
@@ -73,6 +73,50 @@ resource "aws_iam_role" "apiary_task_readwrite" {
 EOF
 }
 
+resource "aws_iam_role_policy" "cloudwatch_for_ecs_readonly" {
+  name = "cloudwatch"
+  role = "${aws_iam_role.apiary_task_readonly.id}"
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "cloudwatch:PutMetricData",
+                "cloudwatch:GetMetricStatistics",
+                "cloudwatch:ListMetrics",
+                "ec2:DescribeTags"
+            ],
+            "Resource": "*"
```

Comment:
Shall we restrict resources based on Tag?

Reply:
@pradeepbhadani I don't see any way to restrict resource for PutMetricData. Other three actions are all readonly actions, not sure if it matters for those?

https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazoncloudwatch.html#amazoncloudwatch-resources-for-iam-policies

Reply:
merging this PR, I also didn't find anyway to restrict this, we can open another issue if we need to tweak.